easybuild/main.py

```diff
@@ -586,9 +490,191 @@ def main(args=None, logfile=None, do_build=None, testing=False, modtool=None):
+
+    # determine easybuild-easyconfigs package install path
+    easyconfigs_pkg_paths = get_paths_for(subdir=EASYCONFIGS_PKG_SUBDIR)
+    if not easyconfigs_pkg_paths:
+        _log.warning("Failed to determine install path for easybuild-easyconfigs package.")
+
+
+   # if EasyStack file is provided, parse it, and loop over the items in the EasyStack file
+   if options.easystack:
+        orig_paths, opts_per_ec = parse_easystack(options.easystack)
+
+        _log.debug("Start build loop over items in the EasyStack file: %s" % orig_paths)
+        # TODO: insert fast loop that validates if all command line options are valid. If there are errors in options,
+        # we want to know early on, and this loop potentially builds a lot of packages and could take very long
+        # for path in orig_paths:
+        #     validate_command_opts(args, opts_per_ec[path])
+        # Loop over each item in the EasyStack file, each time updating the config
+        # This is because each item in an EasyStack file can have options associated with it
+        for path in orig_paths:
+            _log.debug("Starting build for %s" % path)
+            # NOTE: not sure if this is needed. Is the EasyConfigs cache preserved throughout loops of this iteration?
+            # Current 'path' may have different options associated with it. Thus, resolution of EasyConfigs
+            # to full paths should not be read from cache, but redetermined. Thus, we wipe the cache
+            # wipe_cache()  # Wipes the easyconfig cache with _easyconfigs_cache.clear()  # TODO: implement
+
+            # If EasyConfig specific arguments were supplied in EasyStack file
+            # merge arguments with original command line args
+            if path in opts_per_ec:
+                _log.debug("EasyConfig specific options have been specified for "
+                           "%s in the EasyStack file: %s" % (path, opts_per_ec[path]))
+                if args is None:
+                    args = sys.argv[1:]
+                ec_args = dict_to_argslist(opts_per_ec[path])
+                # By appending ec_args to args, ec_args take priority
+                new_args = args + ec_args
+                _log.info("Argument list for %s after merging command line arguments with EasyConfig specific "
+                          "options from the EasyStack file: %s" % (path, new_args))
+            else:
+                # If no EasyConfig specific arguments are defined, sse original args.
+                # That way,set_up_configuration restores the original config
+                new_args = args
+            eb_go, cfg_settings = set_up_configuration(args=new_args, logfile=logfile, testing=testing, reconfigure=True)
+
+            hooks = load_hooks(options.hooks)
+            overall_success = rest_of_main([path], eb_go.options, cfg_settings, modtool, testing, init_session_state,
```

Comment:
Should be something like this:

```python
overall_success = True
if easystack:
  for path in orig_paths
    overall_success &= rest_of_main([path], ...)
else:
    overall_success = rest_of_main(...)
```

And the only do the `if overall_success` + `cleanup` outside of the `if/else`

Reply:
And `overall_success` should be renamed to `do_cleanup` here, since the return value of `rest_of_main` indicates whether or not cleanup should happen.

Reply:
fixed in 8afef99d9af118687955aec402eeaca7db39a5cb